subsys/bluetooth/shell/ias_client.c

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file
+ * @brief Shell APIs for Bluetooth IAS
+ *
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include "sys/printk.h"
+#include <bluetooth/conn.h>
+
+#include <stdint.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/services/ias.h>
+
+#include "bt.h"
+
+extern const struct shell *ctx_shell;
+
+static void ias_discover_cb(struct bt_conn *conn, int err)
+{
+	if (err != 0) {
+		shell_error(ctx_shell, "Failed to discover IAS err: %d\n", err);
+	} else {
+		shell_print(ctx_shell, "IAS discover success\n");
+	}
+}
+
+static struct bt_ias_client_cb ias_client_callbacks = {
+	.discover = ias_discover_cb,
+};
+
+static int cmd_ias_client_init(const struct shell *sh, size_t argc, char **argv)
+{
+	int err;
+
+	if (!ctx_shell) {
+		ctx_shell = sh;
+	}
+
+	err = bt_ias_client_cb_register(&ias_client_callbacks);
+	if (err != 0) {
+		shell_print(sh, "IAS client init failed");
+	} else {
+		shell_print(sh, "IAS client initialized");
+	}
+
+	return err;
+}
+
+static int cmd_ias_client_discover(const struct shell *sh, size_t argc, char **argv)
+{
+	int err;
+
+	err = bt_ias_discover(default_conn);
+
+	if (err != 0) {
+		shell_print(sh, "IAS discover failed");
+	}
+
+	return err;
+}
+static uint8_t bt_ias_alert_lvl(const char *alert_level)
+{
+	if (strcmp(alert_level, "stop") == 0) {
+		return BT_IAS_ALERT_LVL_NO_ALERT;
+	} else if (strcmp(alert_level, "mild") == 0) {
+		return BT_IAS_ALERT_LVL_MILD_ALERT;
+	} else if (strcmp(alert_level, "high") == 0) {
+		return BT_IAS_ALERT_LVL_HIGH_ALERT;
+	}
+
+	return -EINVAL;
```

Comment:
Returning a negative value in a function that is `uint8_t` - Please update the return type to `int`

Reply:
Done